src/utilities/Dealer.cpp

```diff
@@ -19,93 +23,126 @@ Dealer::DealerFlag::DealerFlag(DealerFlagTitle title, DealerFlagPriority priorit
 // Create a distribution of robots according to their flags
 std::unordered_map<std::string, v::RobotView> Dealer::distribute(const std::vector<v::RobotView> &allRobots, const FlagMap &flagMap,
                                                                  const std::unordered_map<std::string, stp::StpInfo> &stpInfoMap) {
-    std::vector<std::vector<double>> scores = getScoreMatrix(allRobots, flagMap, stpInfoMap);
-    std::vector<int> assignment;
-
-    // solve the matrix and put the results in 'assignment'
-    // The cost function will be minimized
-    rtt::Hungarian::Solve(scores, assignment);
-
-    return mapFromAssignments(allRobots, flagMap, assignment);
+    std::unordered_map<std::string, v::RobotView> output;
+    std::vector<RoleScores> scores = getScoreMatrix(allRobots, flagMap, stpInfoMap);
+    // Make index of roles and ID to keep track which are the original indexes of each and get roleNames
+    std::vector<int> indexRoles;
+    std::vector<int> indexID;
+    std::vector<std::string> roleNames;
+    distribute_init(indexRoles,indexID,roleNames,flagMap);
+
+            // Loop through the order of role priorities (column)
+    for (int i = sizeof(DealerFlagPriority); i >= 0; i--){
```

Comment:
maybe add a comment to make clear every priority level can represent a group of robots.

Reply:
I was going to make a good template for every play soon with all this information (bit like the front-end of STP), but i guess the more places the better

Reply:
I was mentioning is because I was a bit confused as first. Also I just realized that sizeof(DealerFlagPriority) is probably a bad idea to do.

Reply:
It may be better to have an explicit order or vector and loop over that (it's also more clear)

Reply:
Dealer.h:
// Lowest to Highest Priority ordering. UNIQUE Classes are the Highest. (Order using in Dealer::distribute() )
enum class DealerFlagPriority { LOW_PRIORITY, MEDIUM_PRIORITY, HIGH_PRIORITY, REQUIRED, KEEPER };

This is the method I implemented the order, the size is allowed to vary aslong as the order is followed like in the comment. How would you suggest it otherwise?